pygsp/graphs/graph.py

```diff
@@ -122,6 +128,246 @@ def __repr__(self, limit=None):
+        -------
+        graph : :class:`~pygsp.graphs.Graph`
+        """
+        import networkx as nx
+        # keep a consistent order of nodes for the agency matrix and the signal array
+        nodelist = graph_nx.nodes()
+        adjacency = nx.to_scipy_sparse_matrix(graph_nx, nodelist, weight=weight)
+        graph = cls(adjacency)
+        # Adding the signals
+        for s_name in signals_name:
+            s_dict = nx.get_node_attributes(graph_nx, s_name)
+            if len(s_dict.keys()) == 0:
+                raise ValueError("Signal {} is not present in the networkx graph".format(s_name))
+            # The signal is set to zero for node not present in the networkx signal
+            s_value = np.array([s_dict[n] if n in s_dict else 0 for n in nodelist])
+            graph.set_signal(s_value, s_name)
+        return graph
+
+    @classmethod
+    def from_graphtool(cls, graph_gt, edge_prop_name='weight', aggr_fun=sum, signals_names=[]):
+        r"""Build a graph from a graph tool object.
+
+        Parameters
+        ----------
+        graph_gt : :py:class:`graph_tool.Graph`
+            Graph tool object
+        edge_prop_name : string
+            Name of the `property <https://graph-tool.skewed.de/static/doc/graph_tool.html#graph_tool.Graph.edge_properties>`_
+            to be loaded as weight for the graph. If the property is not found a graph with default weight set to 1 is created.
+            On the other hand if the property is found but not set for a specific edge the weight of zero will be set
+            therefore for single edge this will result in a none existing edge. If you want to set to a default value please
+            use `set_value <https://graph-tool.skewed.de/static/doc/graph_tool.html?highlight=propertyarray#graph_tool.PropertyMap.set_value>`_
+            from the graph_tool object.
+        aggr_fun : function
+            When the graph as multiple edge connecting the same two nodes the aggragate function is called to merge the
+            edges. By default the sum is taken.
+        signals_names : list[String] or 'all'
+            List of signals names to import from the graph_tool graph or if set to 'all' import all signal present
+            in the graph
+
+        Returns
+        -------
+        graph : :class:`~pygsp.graphs.Graph`
+            The weight of the graph are loaded from the edge property named ``edge_prop_name``
+        """
+        nb_vertex = len(graph_gt.get_vertices())
+        weights = np.zeros(shape=(nb_vertex, nb_vertex))
+
+        props_names = graph_gt.edge_properties.keys()
+
+        if edge_prop_name in props_names:
+            prop = graph_gt.edge_properties[edge_prop_name]
+            edge_weight = prop.get_array()
+        else:
+            warnings.warn("""As the property {} is not found in the graph, a weight of 1.0 is given to each edge"""
+                          .format(edge_prop_name))
+            edge_weight = np.ones(nb_vertex)
```

Comment:
Python style: ask for forgiveness rather than permission. That is:
```
try:
    prop = graph_gt.edge_properties[weight]
    edge_weight = prop.get_array()
except KeyError:
    warnings.warn(...)
    edge_weight = np.ones(nb_vertex)
```